File: Java/src/main/java/Solution901_1000/Solution977.java
package Solution901_1000;

import java.util.Arrays;

/**
 * @author windows
 */
public class Solution977 {

    public int[] sortedSquares(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            nums[i] = nums[i] * nums[i];
        }
        Arrays.sort(nums);
        return nums;
    }


    public static void main(String[] args) {
        Solution977 solution977 = new Solution977();
    }
}

File: C++/src/Solution901_1000/Solution977.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution977.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> sortedSquares(vector<int>& nums) {
  int n = nums.size();
  vector<int> ans(n);
  for (int i = 0, j = n - 1, pos = n - 1; i < j; pos--) {
    if (nums[i] * nums[i] > nums[j] * nums[j]){
      ans[pos] = nums[i] * nums[i];
      i++;
    } else{
      ans[pos] = nums[j] * nums[j];
      j--;
    }
  }
  return ans;
}
